scripts/post_autosave.js

```diff
@@ -0,0 +1,164 @@
+﻿/*
+ TODO:
+
+ Trigger removal of cached posts off successful posting instead of form submission (in case something goes wrong)
+*/
+settingsLoadedEvent.addHandler(function() {
+    if (getSetting("enabled_scripts").contains("post_autosave")) {
+        PostAutosave = {
+            CACHE_SETTING_NAME: 'autosaved_posts',
+
+            // Max number of posts cached
+            CACHE_SIZE: 10,
+
+            // The amount of time (ms) after typing before posts are cached 
+            CACHE_TIME: 500,
+
+            // Local copy of the cache to use within a single page view
+            cache: [],
```

Comment:
Order of the cache entries matters since I want recently updated to drafts to "refresh" by moving to the front.  Once the max number of entries is hit, I truncate the array and all but the last 10 drafts touched will be removed.  Storing this as an array makes these things easy due to the methods available for arrays, at the expense of having to loop through it.  Changing it to an object would bring its own complications.